### PR TITLE
chore: add spread operator, refactor tests

### PR DESCRIPTION
**Pull request by bshaffer**

### CODE REVIEW

File: tests/Tests/Unit/ArrayTraitTest.php [hunk 1]
```diff
@@ -50,7 +50,7 @@ public function testPluck()
         $value = '123';
         $key = 'key';
         $array = [$key => $value];
-        $actualValue = $this->implementation->call('pluck', [$key, &$array]);
+        $actualValue = $this->implementation->pluck($key, $array);
```
Comment (vishwarajanand) [inline tests/Tests/Unit/ArrayTraitTest.php:53]:
```suggestion
        $actualValue = $this->implementation->pluck($key, &$array);
```

File: tests/Tests/Unit/ArrayTraitTest.php [hunk 2]
```diff
@@ -60,7 +60,7 @@ public function testPluckThrowsExceptionWithInvalidKey()
     {
         $array = [];
         $this->expectException(InvalidArgumentException::class);
-        $this->implementation->call('pluck', ['not_here', &$array]);
+        $this->implementation->pluck('not_here', $array);
```
Comment (vishwarajanand) [inline tests/Tests/Unit/ArrayTraitTest.php:63]:
```suggestion
        $this->implementation->pluck('not_here', &$array);
```

File: tests/Tests/Unit/ArrayTraitTest.php [hunk 3]
```diff
@@ -72,25 +72,25 @@ public function testPluckArray()
         ];
         $expectedArray = $array;
 
-       $actualValues = $this->implementation->call('pluckArray', [$keys, &$array]);
+       $actualValues = $this->implementation->pluckArray($keys, $array);
```
Comment (vishwarajanand) [inline tests/Tests/Unit/ArrayTraitTest.php:75]:
```suggestion
       $actualValues = $this->implementation->pluckArray($keys, &$array);
```

**Review by vishwarajanand — APPROVED** [review]
Lgtm

File: tests/Tests/Unit/ArrayTraitTest.php (hunk 1 above)
Comment (bshaffer) [reply tests/Tests/Unit/ArrayTraitTest.php:53]: It's not necessary to supply the `&` on this side, since it's part of the `pluck` method signature already

Comment (bshaffer) [reply tests/Tests/Unit/ArrayTraitTest.php:53]: More information: https://www.php.net/manual/en/language.references.pass.php

> Note: There is no reference sign on a function call - only on function definitions. Function definitions alone are enough to correctly pass the argument by reference.

And here as well: https://stackoverflow.com/questions/8971261/php-5-4-call-time-pass-by-reference-easy-fix-available

Call-time pass-by-reference was deprecated in PHP 5.3 and removed entirely in PHP 5.4

**Pull request merged**